### PR TITLE
Added docker image promotion api

### PR DESCRIPTION
**Pull request by yuvalsimon**
Usage example:
```
from artifactory import ArtifactoryPath

path = ArtifactoryPath("http://example.com/artifactory")
path.promote_docker_image("docker-staging", "docker-prod", "my-application", "0.5.1")
```
Promotes the image myapplciation:0.5.1 from staging registry to prod registry

### CODE REVIEW

File: artifactory.py [hunk 1]
```diff
@@ -1693,6 +1693,43 @@ def from_aql(self, result):
...
+        promote_data = json.dumps(
+            {
+                "targetRepo": target_repo,
+                "dockerRepository": docker_repo,
+                "tag": tag,
+                "copy": copy,
+            }
+        )
+        headers = {"Content-Type": "application/json"}
+        r = self.session.post(promote_url, data=promote_data, headers=headers)
```
Comment (allburov) [inline artifactory.py:1720]: `requests` library can work with a dict
```python
self.session.post(promote_url, json=promote_data)
```

**Comment by allburov** [comment]
Could you add an example of usage to README? Just as you did in the description

**Comment by allburov** [comment]
@yuvalsimon Waiting for an example to README and we're good to merge it!

**Comment by yuvalsimon** [comment]
Missed your comment. Done

**Pull request merged**